### PR TITLE
UIU-2801 Assign/unassign a user affiliations

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose
https://issues.folio.org/browse/UIU-2801

Add the ability to manage user assignments through a modal.

Consortia API - https://s3.amazonaws.com/foliodocs/api/mod-consortia/s/consortia.html
Consortium tenants API - https://s3.amazonaws.com/foliodocs/api/mod-consortia/s/tenants.html
Consortium user-tenant association API - https://s3.amazonaws.com/foliodocs/api/mod-consortia/s/user_tenants.html

## Approach

- Implement modal components to assign, unassign, sort, and filter user affiliations (`<AffiliationsManager>`, `<AffiliationsManagerModal>`, `<AffiliationsManagerModalFooter>`, `<AffiliationsManagerFiltersPane>`, `<AffiliationsManagerResultsPane>`, `<AffiliationsManagerTrigger>`);
- Implement hook for affiliation assignment management (`useAffiliationsAssignment`) - responsible for selecting and unselecting items in the modal;
- Implement hook to get consortium tenants (`useConsortiumTenants`);
- Implement hook to update user tenants associations (`useUserAffiliationsMutation`) - send POST for newly assigned tenants and DELETE for newly unassigned ones;
- Cover new logic with unit tests;

<details open><summary>Preview</summary>
<p>

chrome_gCHMOUTaS2

</p>
</details>

### CODE REVIEW

File: src/components/AffiliationsManager/AffiliationsManager.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@import "@folio/stripes-components/lib/variables";
```
Comment (artem-blazhko) [inline src/components/AffiliationsManager/AffiliationsManager.css:1]: What is the reason to import this style package?

File: src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModalFooter.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      </Button>
+
+      <div>
+        <FormattedMessage
+          id="ui-users.affiliations.manager.modal.totalSelected"
+          values={{ count: totalSelected }}
+        />
+      </div>
+
+      <Button
```
Comment (artem-blazhko) [inline src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModalFooter.js:31]: Could you remove empty lines here?

File: src/components/AffiliationsManager/AffiliationsManagerResultsPane/AffiliationsManagerResultsPane.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      paneSub={paneSub}
+      firstMenu={firstMenu}
+    >
+      <MessageBanner
+        show={displayWarning}
+        type="warning"
+      >
+        <FormattedMessage id="ui-users.affiliations.manager.modal.warning.unassigned" />
+      </MessageBanner>
+
```
Comment (artem-blazhko) [inline src/components/AffiliationsManager/AffiliationsManagerResultsPane/AffiliationsManagerResultsPane.js:108]: remove this empty line

File: src/components/AffiliationsManager/AffiliationsManager.css (hunk 1 above)
Comment (usavkov-epam) [reply src/components/AffiliationsManager/AffiliationsManager.css]: Oh, it's really redundant here, thanks! I have already removed that

File: src/components/AffiliationsManager/AffiliationsManagerFiltersPane/AffiliationsManagerFiltersPane.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  Pane,
+  PaneMenu,
+  SearchField,
+} from '@folio/stripes/components';
+import { CollapseFilterPaneButton } from '@folio/stripes/smart-components';
+
+import {
+  AFFILIATIONS_FILTERS_ACTION_TYPES,
+  SEARCH_FIELD_NAME
+} from '../constants';
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/components/AffiliationsManager/AffiliationsManagerFiltersPane/AffiliationsManagerFiltersPane.js]: Good catch! I've gone through the code again and updated some places.

File: src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModalFooter.js (hunk 2 above)
Comment (usavkov-epam) [reply src/components/AffiliationsManager/AffiliationsManagerModal/AffiliationsManagerModalFooter.js]: Done

File: src/components/AffiliationsManager/AffiliationsManagerResultsPane/AffiliationsManagerResultsPane.js (hunk 3 above)
Comment (usavkov-epam) [reply src/components/AffiliationsManager/AffiliationsManagerResultsPane/AffiliationsManagerResultsPane.js]: Done

File: src/constants.js [hunk 5]
```diff
@@ -313,5 +313,19 @@ export const STATUS_CODES = {
   UNPROCESSABLE_ENTITY: 422,
 };
 
+export const LIMIT_MAX = 1000;
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/constants.js]: Actually, we can, thanks for noticing that! Updated

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**